### DISCUSSION TITLE
[General] How to upgrading zksync era?

### DISCUSSION THREAD

**Discussion by Niet-Pham** [zkSync-Community-Hub/zksync-developers#402] [ZK Stack] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

My custom zksync era was built at tags v6.0.0-rc1. Can able to I upgrade from v6.0.0-rc1 to tags core-v21.0.0? Or do I need to upgrade from v6.0.0-rc1 -> v7.0.0-rc0 -> .... -> core-v21.0.0? Btw can anyone provide me documents for upgrading the network zksync?

### What is the specific issue or error you're encountering?

I don't find any upgrade docs

### Can you share the error messages or logs you're receiving, if any?

I tried upgrade via update new zksync contract facets (Mailbox, DiamondCut,..) but don't update any code just deploy new contracts and upgrade. Then when I run `zk server` got error at here
![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/55547523/2e708657-8e98-460d-960f-79ec6015434b)

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

![image](https://github.com/zkSync-Community-Hub/zksync-developers/assets/55547523/300669b9-135e-4816-9324-72d5d1bf85fa)

### Have you tried to isolate the problem, and if so, what were the results?

I think when I update new facts. when run `zk server` it read event upgrade and decode ev but got empty array on it

### What steps have you already taken to try to resolve the issue?

I print log in code

**Comment by iscxyz** [comment, marked as answer]
Hello, @Niet-Pham - we are working on the tools to smoother upgrades, but right now the best approach in the short term is redeploy if possible.
